Form/View/ImporterViewType.php

```diff
@@ -0,0 +1,140 @@
+
+use Elasticsearch\Client;
+use EMS\CommonBundle\Helper\EmsFields;
+use EMS\CoreBundle\Entity\View;
+use EMS\CoreBundle\Form\Nature\ImporterType;
+use EMS\CoreBundle\Service\FileService;
+use EMS\CoreBundle\Service\JobService;
+use Psr\Log\LoggerInterface;
+use Symfony\Bundle\FrameworkBundle\Routing\Router;
+use Symfony\Component\Form\Extension\Core\Type\CheckboxType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormFactory;
+use Symfony\Component\Form\FormFactoryInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+use Twig_Environment;
+
+/**
+ * It's the mother class of all specific DataField used in eMS
+ *
+ * @author Mathieu De Keyzer <ems@theus.be>
+ *
+ */
+class ImporterViewType extends ViewType
+{
+
+    /**
+     * @var FileService
+     */
+    private $fileService;
+    /**
+     * @var JobService
+     */
+    private $jobService;
+    /**
+     * @var TokenStorageInterface
+     */
+    private $security;
+    /**
+     * @var Router
+     */
+    private $router;
+
+    public function __construct(FormFactory $formFactory, Twig_Environment $twig, Client $client, LoggerInterface $logger, FileService $fileService, JobService $jobService, TokenStorageInterface $security, Router $router)
+    {
+        parent::__construct($formFactory, $twig, $client, $logger);
+        $this->fileService = $fileService;
+        $this->jobService = $jobService;
+        $this->security = $security;
+        $this->router = $router;
+    }
+
+    public function getLabel()
+    {
+        return "Importer: form to import a zip file containing JSON files";
+    }
+
+    public function getName()
+    {
+        return "Importer";
+    }
+
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        parent::buildForm($builder, $options);
+        $builder
+            ->add('finalize', CheckboxType::class, [
+                'data' => true,
+                'required' => false,
+            ])
+            ->add('signData', CheckboxType::class, [
+                'data' => true,
+                'required' => false,
+            ])
+            ->add('businessKey', CheckboxType::class, [
+                'data' => false,
+                'required' => false,
+            ])
+            ->add('rawImport', CheckboxType::class, [
+                'data' => false,
+                'required' => false,
+            ])
+            ->add('force', CheckboxType::class, [
+                'data' => false,
+                'required' => false,
+            ]);
+    }
+
+
+    public function getParameters(View $view, FormFactoryInterface $formFactory, Request $request)
+    {
+        return [];
+    }
+
+    public function generateResponse(View $view, Request $request)
```

Comment:
```suggestion
    public function generateResponse(View $view, Request $request): ResponseInterface
```

Reply:
4159cf31192b018363450c77cd4b901ffc17b2d3

Reply:
Great yes ResponseInterface does not exists, it's just Response